packages/openai-compatible/src/openai-compatible-chat-language-model.ts

```diff
@@ -0,0 +1,537 @@
+
+type OpenAICompatibleChatConfig = {
+  provider: string;
+  headers: () => Record<string, string | undefined>;
+  url: (options: { modelId: string; path: string }) => string;
+  fetch?: FetchFunction;
+};
+
+export class OpenAICompatibleChatLanguageModel implements LanguageModelV1 {
+  readonly specificationVersion = 'v1';
+
+  readonly supportsStructuredOutputs = false;
+
+  readonly modelId: OpenAICompatibleChatModelId;
+  readonly settings: OpenAICompatibleChatSettings;
+
+  private readonly config: OpenAICompatibleChatConfig;
+
+  constructor(
+    modelId: OpenAICompatibleChatModelId,
+    settings: OpenAICompatibleChatSettings,
+    config: OpenAICompatibleChatConfig,
+  ) {
+    this.modelId = modelId;
+    this.settings = settings;
+    this.config = config;
+  }
+
+  get defaultObjectGenerationMode(): 'json' | 'tool' | undefined {
+    return this.settings.defaultObjectGenerationMode;
+  }
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  private getArgs({
+    mode,
+    prompt,
+    maxTokens,
+    temperature,
+    topP,
+    topK,
+    frequencyPenalty,
+    presencePenalty,
+    stopSequences,
+    responseFormat,
+    seed,
+    stream,
+  }: Parameters<LanguageModelV1['doGenerate']>[0] & {
+    stream: boolean;
+  }) {
+    const type = mode.type;
+
+    const warnings: LanguageModelV1CallWarning[] = [];
+
+    if (topK != null) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'topK',
+      });
+    }
+
+    if (
+      responseFormat != null &&
+      responseFormat.type === 'json' &&
+      responseFormat.schema != null
+    ) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'responseFormat',
+        details: 'JSON response format schema is not supported',
+      });
+    }
+
+    const baseArgs = {
+      // model id:
+      model: this.modelId,
+
+      // model specific settings:
+      user: this.settings.user,
+
+      // standardized settings:
+      max_tokens: maxTokens,
+      temperature,
+      top_p: topP,
+      frequency_penalty: frequencyPenalty,
+      presence_penalty: presencePenalty,
+      stop: stopSequences,
+      seed,
+
+      // response format:
+      response_format:
+        // TODO(shaper): Review vs. OpenAI impl here.
+        // json object response format is not currently supported
+        undefined,
+
+      // messages:
+      messages: convertToOpenAICompatibleChatMessages(prompt),
+    };
+
+    switch (type) {
+      case 'regular': {
+        const { tools, tool_choice, toolWarnings } = prepareTools({ mode });
+        return {
+          args: {
+            ...baseArgs,
+            tools,
+            tool_choice,
+          },
+          warnings: [...warnings, ...toolWarnings],
+        };
+      }
+
+      case 'object-json': {
+        // TODO(shaper): Review vs. OpenAI impl here.
+        throw new UnsupportedFunctionalityError({
+          functionality: 'object-json mode',
+        });
```

Comment:
copy what openai has including schemas

Reply:
Done, I omitted `structuredOutputs` since AFAIK it's only supported by OpenAI today.